RELEASE.rst

```diff
@@ -31,39 +31,74 @@ that changes the major version number.
 Doing a Release
 ---------------
 
-1. Test releases on Python 2.7 and 3.4+ on Windows, Linux and OSX,
-   with and without the C extensions. It's generally enough to just run the
-   tests on 2.7, 3.4 and the latest 3.x version with and without the
-   extensions on a single platform, and then just test any version on the
-   other platforms as a sanity check. `python setup.py test` will build the
-   extensions and test. `python tools/clean.py` will remove the extensions,
+1. PyMongo is tested on Evergreen. Ensure the latest commit are passing CI
+   as expected: https://evergreen.mongodb.com/waterfall/mongo-python-driver.
+   To test locally, `python setup.py test` will build the C extensions and
+   test. `python tools/clean.py` will remove the extensions,
    and then `python setup.py --no_ext test` will run the tests without
    them. You can also run the doctests: `python setup.py doc -t`.
 
-2. Add release notes to doc/changelog.rst. Generally just summarize/clarify
+2. Check Jira to ensure all the tickets in this version have been completed.
+
+3. Add release notes to doc/changelog.rst. Generally just summarize/clarify
    the git log, but you might add some more long form notes for big changes.
 
-3. Search and replace the "devN" version number w/ the new version number (see
+4. Search and replace the "devN" version number w/ the new version number (see
```

Comment:
What note is being referred to here?

Reply:
I assume it's the note in Versioning. I added a link.